packages/documentation/src/pages/architecture/unit-tests.mdx

```diff
@@ -0,0 +1,120 @@
+## How do unit tests work?
+
+#### Where they live
+- Tests can live anywhere in the `src` directory, but in general are located in a directory close to the code or application that they are written to test.
+- The `.unit.spec.js` file extension is used for unit tests, while `.e2e.spec.js` is used for end-to-end (E2E) tests.
+- Visual regression test config has its own directory under `src/platform/testing/visual-regression`
+- Test commands are specified in `package.json`, as well as in vets-website's `README.md`
+- Configuration files for nightwatch and sass-lint live under `config/`
+- Scripts for running nightwatch tests live under `script/`
+
+#### When they run/ how they run
+There are 5 times tests can run:
+
+  1. At any time during local development by running one of the npm scripts specified in `package.json`
+    - Any tests can be run during this time
+  2. During the pre-commit hook
+    - Only linting is run automatically
+  3. Before merging a branch to master via Jenkins
+    - Unit tests, e2e tests, and code climate are automatically run during this time
+  4. After merging to master
+    - Accessibility tests are automatically run during this time
+    - Unit tests and e2e tests are run automatically
+
+#### How to debug
+For unit tests:
+- You can put a console.log in your test code. When you run the test, the output will be in the command line.
+- Use Chrome devtools to debug unit tests
+- VSCode's (or any other editor's) debugging tool for mocha tests
+- Karma has some pretty great tools for running your tests in a browser so that you can set breakpoints and see the test execute. But we're not using Karma in vets-website, so if we wanted this functionality we would have to add it.
+
+For e2e tests:
+- It's possible to set an infinite pause (`.pause()`) in the test code and then see what's happening at a given point in the browser where the test is running. It is also possible to use the Chrome devtools and inspect what's going on (this is only possible when the browser is Chrome and not Electron, so you would need to disable Electron is `config/nightwatch.js`).
+
+
+#### When we write them
+Write tests as you go! New functionality added in a PR should ideally be covered by unit and e2e tests, where applicable.
+
+
+### Libraries
+
+1. Test environment
+- **mocha**: testing framework
```

Comment:
not sure how @jbalboni feels about this but I think we should refer to Mocha as a `test running framework`- which differentiates it from an assertion library in my mind. 

It might be good to write a small blurb about what this means in terms of is responsibilities. maybe something like: finds the test files and builds and runs the test suite